lib/screens/v2/candidate_generator/elevator/closures.ex

```diff
@@ -65,14 +65,33 @@ defmodule Screens.V2.CandidateGenerator.Elevator.Closures do
     active_closures = Enum.flat_map(active, &elevator_closure/1)
     at_this_elevator? = fn %Closure{id: id} -> id == elevator_id end
 
+    {:ok, %Stop{id: stop_id}} = @facility.fetch_stop_for_facility(elevator_id)
+
+    relevant_closures =
+      Enum.filter(active_closures, &relevant_closure?(&1, stop_id, active_closures))
+
     case Enum.find(active_closures, at_this_elevator?) do
       nil ->
         upcoming_closures =
           upcoming |> Enum.flat_map(&elevator_closure/1) |> Enum.filter(at_this_elevator?)
 
+        footer =
+          if relevant_closures ++ upcoming_closures == [], do: [], else: [%Footer{screen: config}]
```

Comment:
The `++` operator has to iterate through the entire left-hand side, so using a pattern like this to check for emptiness can be slow when the list is large (relatively speaking; it could probably have thousands of items before you'd really notice, since list iteration itself is extremely fast). The more idiomatic way to handle this would be `relevant_closures == [] and upcoming_closures == []`, or possibly `Enum.empty?(relevant_closures) and Enum.empty?(upcoming_closures)` if that reads better to you.